gap/grahom.gi

```diff
@@ -872,3 +872,104 @@ function(D, t)
   n := DigraphNrVertices(D);
   return IsDigraphColouring(D, ImageListOfTransformation(t, n));
 end);
+
+InstallMethod(MaximalCommonSubdigraph, "for a pair of digraphs",
+[IsDigraph, IsDigraph],
+function(D1, D2)
+  local B, out1, out2, adj, good, p, Clqus, M, l,
+        embedding1, embedding2, iso;
+
+  iso := IsomorphismDigraphs(D1, D2);
+
+  if iso <> fail then
+    return [DigraphCopy(D1), IdentityTransformation, AsTransformation(iso)];
+  fi;
+
+  out1 := OutNeighbours(D1);
+  out2 := OutNeighbours(D2);
+
+  adj := function(pair)
+    local L, isadj;
+    isadj := function(p)
+      if p[1] = pair[1] or p[2] = pair[2] then
+        return false;
+      fi;
+      return Number(out1[pair[1]], x -> x = p[1]) =
+             Number(out2[pair[2]], x -> x = p[2]);
+    end;
+    L := Filtered(Cartesian([1 .. Size(out1)], [1 .. Size(out2)]), isadj);
+    return List(L, y -> ((y[1] - 1) * Size(out2) + (y[2] - 1)) + 1);
+  end;
+
+  B := Digraph(List(Cartesian([1 .. Size(out1)], [1 .. Size(out2)]), adj));
+
+  good := function(v)
+    p := [QuoInt(v - 1, Size(out2)) + 1, RemInt(v - 1, Size(out2)) + 1];
+    return Number(out1[p[1]], x -> x = p[1]) =
+           Number(out2[p[2]], x -> x = p[2]);
+  end;
+
+  B := InducedSubdigraph(B, Filtered(DigraphVertices(B), good));
+  if DigraphNrVertices(B) = 0 then
+    return [Digraph([]), [], []];
+  fi;
+
+  Clqus := DigraphMaximalCliquesReps(B);
+
+  if Clqus = [] then
+    return [Digraph([]), [], []];
```

Comment:
Would there potentially be an interest in getting the second and third list elements to be transformations in this case? I'm not sure what you think of returning `IdentityTransformation` rather than `[]` if the maximal common subdigraph has no vertices; but that way we can guarantee that this function will always output a list which has transformations as second and third elements. For example,
```gap
gap> comm := MaximalCommonSubdigraph(D1, D2);;
gap> 2 ^ comm[2];
```
will never return any errors.

Reply:
I agree, I'll fix this this thanks.